firedantic/_async/model.py

```diff
@@ -53,6 +69,14 @@ def _get_col_ref(cls, name: Optional[str]) -> AsyncCollectionReference:
     return collection
 
 
+def collection_index(*fields: IndexField) -> IndexDef:
```

Comment:
The `collection_index` and `collection_group_index` are now defined twice with the exact same structure in both `_async.models` and in `_sync.models` and then imported in the main `__init__.py` from the async version. Wondering that would it make sense to move them (as well as `IndexField` and `IndexDef`) out from the `_async`/`_sync` directories to reduce duplication and make the import a bit more clear in the main `__init__.py`, so nobody wonders why we're importing them from the async version only.